lib/media/segment_index.js

```diff
@@ -42,14 +42,49 @@ shaka.media.SegmentIndex = class {
      * The number of references that have been removed from the front of the
      * array.  Used to create stable positions in the find/get APIs.
      *
-     * @protected {number}
+     * @private {number}
```

Comment:
I'm not sure why you changed this to be private instead of protected, and added protected setter/getter.  But this could break any subclasses already created by applications, so please revert it.